lib/lotus/model/migrator/connection.rb

```diff
@@ -0,0 +1,83 @@
+module Lotus
+  module Model
+    module Migrator
+      # Sequel connection wrapper
+      #
+      # Normalize external adapters interfaces
+      #
+      # @since 0.4.1
+      # @api private
+      class Connection
+        attr_reader :adapter_connection
+
+        def initialize(adapter_connection)
+          @adapter_connection = adapter_connection
+        end
+
+        def opts
+          @adapter_connection.opts
+        end
+
+        # @since 0.4.1
+        # @api private
+        def host
+          @host ||= opts.fetch(:host, parsed_uri.host)
+        end
+
+        # @since 0.4.1
+        # @api private
+        def port
+          @port ||= opts.fetch(:port, parsed_uri.port)
+        end
+
+        # @since 0.4.1
+        # @api private
+        def database
+          @database ||= opts.fetch(:database, parsed_uri.path[1..-1])
+        end
+
+        # @since 0.4.1
+        # @api private
+        def database_type
+          @adapter_connection.database_type
+        end
+
+        # @since 0.4.1
+        # @api private
+        def user
+          @user ||= opts.fetch(:user, parsed_uri.to_s.match(/[\?|\&]user=(\w+)\&?/).to_a.last)
+        end
+
+        # @since 0.4.1
+        # @api private
+        def password
+          @password ||= opts.fetch(:password, parsed_uri.to_s.match(/[\?|\&]password=(\w+)\&?/).to_a.last)
```

Comment:
`parsed_uri.to_s.match(/[\?|\&]password=(\w+)\&?/).to_a.last` can be moved to a private method